Utils/src/FastSimWeightPR31285To36122.cc

```diff
@@ -0,0 +1,246 @@
+// -*- C++ -*-
+// Package:    TreeMaker
+// Class:      FastSimWeightPR31285To36122
+// Authors:   Sam Bein 
+//         Created:  Wed March 7, 2014
+//         Modified: Thurs March 3, 2021
+
+#include <memory>
+#include <iostream>
+#include <unordered_set>
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include <DataFormats/HepMCCandidate/interface/GenParticle.h>
+#include "DataFormats/JetReco/interface/GenJet.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "TLorentzVector.h"
+
+#include <vector>
+
+class FastSimWeightPR31285To36122 : public edm::global::EDProducer<> {
+
+public:
+    explicit FastSimWeightPR31285To36122(const edm::ParameterSet&);
+    ~FastSimWeightPR31285To36122() override;
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+    void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+    // ----------member data ---------------------------
+
+    edm::InputTag genCollection;
+    edm::EDGetTokenT<edm::View<reco::GenParticle>> genCollectionTok;
+    
+    edm::InputTag genJetTag;
+    edm::EDGetTokenT<edm::View<reco::GenJet>> genJetTok;    
+    edm::InputTag recJetTag;
+    edm::EDGetTokenT<edm::View<pat::Jet>> recJetTok;   
+    
+    
+   TH1F *hRatio_GenJetHadronPtGenJetHadronFlavorLt4;
+   TH1F *hRatio_GenJetHadronPtGenJetHadronFlavorEqEq4;
+   TH1F *hRatio_GenJetHadronPtGenJetHadronFlavorEqEq5;
+   TAxis * xax;
+   
```

Comment:
remove unnecessary whitespace/linebreaks (throughout this file)

Reply:
done